### PR TITLE
REFACTOR: Mecha icon_state

### PR DESCRIPTION
**Pull request by ErdGinalD**
## What Does This PR Do
Refactors the mech's sprite update code, unifying the code to fit any change to the starting sprite, allowing you to freely add any future skin on a mech without fear of sprite update bugs or errors.

## Why It's Good For The Game
Refactor is good. Also let paintkits on rupara working without ussues and bugs, or here if find any good in future.

## Images of changes
change completely nothing, but let code more variable for future.

## Testing
all tested in rupara with local paintkits. 2 paintkits here also work completely smooth

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -772,14 +772,14 @@
 			return
 
 		user.visible_message("[user] opens [P] and spends some quality time customising [src].")
+		if(do_after_once(user, 3 SECONDS, target = src))
+			name = P.new_name
+			desc = P.new_desc
+			initial_icon = P.new_icon
+			reset_icon()
```
Comment (DGamerL) [inline code/game/mecha/mecha.dm:780]:
```suggestion
```

**Comment by Henri215** [comment]
Changelog can be removed, this is a non player facing change.

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -1531,6 +1530,7 @@
 			AI = occupant
 			occupant = null
 		var/obj/structure/mecha_wreckage/WR = new wreckage(loc, AI)
+		WR.icon_state = "[src.reset_icon(loc, AI)]-broken"
```
Comment (Sirryan2002) [inline code/game/mecha/mecha.dm:1533]:
```suggestion
		WR.icon_state = "[reset_icon(loc, AI)]-broken"
```

**Comment by Sirryan2002** [comment]
Labeled with permission of headcoders

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**